src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
```

Comment:
Spoke with Manish offline. For 1, we should ping the CLR folks to make sure this was an intended change because this will break a lot of code in the wild and needs to be done intentionally. 
For 2, It's bizarre that Codebase is set in VS but not in the commandline. Manish is going to investigate a bit more to see why that's happening and if setting that to null would break pdb resolution for example.

This change is very simple and correct and so let's merge this change and then follow up on 1 and 2.